plugins/plugin-core-sfuzz/lib/SfuzzObjectiveManager.ts

```diff
@@ -37,55 +39,67 @@ export class SfuzzObjectiveManager<
   T extends Encoding
 > extends StructuralObjectiveManager<T> {
   /**
-   * Constructor.
-   *
-   * @param runner Encoding runner
+   * @inheritdoc
    */
-  constructor(runner: EncodingRunner<T>) {
-    super(runner);
-  }
-
-  /**
-   * @inheritDoc
-   * @protected
-   */
-  protected _updateObjectives(
-    objectiveFunction: ObjectiveFunction<T>,
+  public async evaluateOne(
     encoding: T,
-    distance: number
-  ): void {
-    // When objective is covered update objectives
-    if (distance === 0.0) {
-      // Remove objective from the current and uncovered objectives
-      this._uncoveredObjectives.delete(objectiveFunction);
-      this._currentObjectives.delete(objectiveFunction);
+    budgetManager: BudgetManager<T>,
+    // eslint-disable-next-line @typescript-eslint/no-unused-vars
+    terminationManager: TerminationManager
+  ): Promise<void> {
+    // Execute the encoding
+    const result = await this._runner.execute(this._subject, encoding);
+    budgetManager.evaluation(encoding);
+
+    // Store the execution result in the encoding
+    encoding.setExecutionResult(result);
 
-      // Add objective to the covered objectives and update the archive
-      this._coveredObjectives.add(objectiveFunction);
-      if (!this._archive.has(objectiveFunction)) {
-        this._archive.update(objectiveFunction, encoding);
+    // For all current objectives
+    this._currentObjectives.forEach((objectiveFunction) => {
+      // Calculate and store the distance
+      const distance = objectiveFunction.calculateDistance(encoding);
+      if (distance > 1) {
+        // This is to ignore the approach level
+        encoding.setDistance(objectiveFunction, 1);
```

Comment:
This might cause a problem with the current way of calculating the approach level. Since the approach level is now calculated from the chile node of the branch (#175), the approach level is always above one if the branch is not covered already.

Reply:
I actually fix this problem in #256. Once that PR is merged the approach level will actually be zero.